OloEngine/tests/Rendering/MeshOptimizationTest.cpp

```
@@ -0,0 +1,505 @@
+    auto lod = MeshOptimization::GenerateLODMesh(*mesh, 0.01f, 1.0f);
+    ASSERT_NE(lod, nullptr);
+
+    // Should produce at least 1 triangle
+    EXPECT_GE(lod->GetIndices().Num(), 3);
+}
+
+// =============================================================================
+// Shadow Index Buffer Tests
+// =============================================================================
+
+TEST(MeshOptimization, GenerateShadowIndicesProducesSameCount)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)